norduser/utmp.go

```diff
@@ -0,0 +1,94 @@
+package norduser
+
+/*
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <utmp.h>
+
+const int ERROR_REALLOC = -1;
+const int ERROR_MALLOC_USERNAME = -2;
+
+void free_users_table(char*** users, int size) {
+  for (int index = 0; index < size; index++) {
+    if ((*users)[index] != NULL) {
+      free((*users)[index]);
+    }
+  }
+
+  if (*users != NULL) {
+    free(*users);
+  }
+}
+
+int get_utmp_user_processes(char*** users) {
+  int index = 0;
+  int size = 0;
+  setutent();
+
+  *users = NULL;
+
+  struct utmp* u;
```

Comment:
In Python it is possible to define C structs using Python's ctypes library and then make calls to C functions populating those struct directly from Python. The benefit is that you don't have to write and compile C modules and you can use automatic memory management provided by Python, instead manually calling `malloc` etc. Isn't something similar possible in Go?

Reply:
I couldn't find anything like that. You can call C functions directly in go but you still have to manage the memory.